data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/InjectionSetpointAdderImpl.java

```diff
@@ -46,13 +48,19 @@ public InjectionSetpointAdder withSetpoint(double setpoint) {
         return this;
     }
 
+    @Override
+    public InjectionSetpointAdder withUnit(Unit unit) {
+        this.unit = unit;
+        return this;
+    }
+
     @Override
     public NetworkActionAdder add() {
```

Comment:
check here that if the unit is section_count, that the setpoint is a positive integer. otherwise raise a FaraoException

Reply:
Done !